### PR TITLE
make protoc version, path configurable

### PR DESCRIPTION
**Pull request by mgfcmartinez**
Add a protoc config section to allow users to pin a specific protoc
version. Make the path configurable as well- gunk will attempt to
execute the binary at the path, or if none exists, gunk will download
the binary to this path.

Add a subcommand to the download command. The user now has to specify
that they want to download protoc. This makes the download command
more flexible, for when we want to use it to download other
protoc-related tools. Allow users to specify the path and version
with the download command as well.

Update the README to reflect the changes and fix miscellaneous
formatting problems while at it.

[OB-367] #done

### CODE REVIEW

File: generate/download.go [hunk 1]
```diff
@@ -104,10 +111,40 @@ func CheckOrDownloadProtoc() (string, error) {
...
+
+		if err := verifyProtocBinary(dstPath, version); err != nil {
+			return "", err
+		}
+
 		return dstPath, nil
 	}
 	return "", fmt.Errorf("unable to download and extract protoc")
 }
+func verifyProtocBinary(path, version string) (err error) {
```
[the remark this answers is not in the document]
Comment (mgfcmartinez) [reply generate/download.go]: my bad, initially i did something with it in defer but i got rid of that block. forgot to update the signature

**Comment by mgfcmartinez** [comment]
@mvdan 

> What will this do if I'm pinning v1.0, download protoc, change the config to pin 1.1, and download again?

this will work as expected, provided they aren't setting the path as well. or if they do set the path, it should be different from the one used by the previous binary:

``` ini
# old
[protoc]
version=v1.0
path=/foo/protoc-v1.0

# new (updated path)
[protoc]
version=v1.1
path=/foo/protoc-v1.1
```

otherwise i'm not sure how to distinguish a version bump while keeping the binary at the same location from just plain old having the wrong version at that path

File: .travis.yml [hunk 2]
```diff
@@ -11,5 +11,5 @@ env:
 install: true
 
 script:
-  - go run main.go download
+  - go run main.go download protoc
```
[the remark this answers is not in the document]
Comment (mgfcmartinez) [reply .travis.yml]: the command originally really only did download protoc, but you're right- need to add a way to download all the required tools. i added an `all` command to download. unfortunately [optional subcommands](https://github.com/alecthomas/kingpin/issues/286) aren't available on kingpin.v2, so i can't get the desired behavior with just `gunk download`

**Comment by mvdan** [comment]
I'm not sure that we want to support multiple sections of `[protoc]` at once. The tool will only use one, so I don't see a reason to do anything fancier.

In any case, assuming that an existing file is the version we want is okay as a start. Something we could do is run `--version` on that path, and redownload the version if it's not the same one - but that can happen as a follow-up PR.

Let me know when this is ready for a final review and we can merge soon after.

**Comment by mgfcmartinez** [comment]
sorry, could've written that example better. what i meant was, if you take the ff. gunkconfig:

```ini
[protoc]
version=v1.0
```

and change it to:

```ini
[protoc]
version=v1.1
```

it will work as expected- downloading the updated proto version. however, if the path was specified but identical to the old one: 

```ini
[protoc]
version=v1.0
path=/protoc
```

and change it to:

```ini
[protoc]
version=v1.1
path=/protoc
```

you'll get something like `error: want protoc version "v1.1" got "v1.0"'`

**Pull request merged**